core/trino-spi/src/main/java/io/trino/spi/classloader/ThreadContextClassLoader.java

```diff
@@ -13,9 +13,24 @@
  */
 package io.trino.spi.classloader;
 
+import java.io.Closeable;
+
 public class ThreadContextClassLoader
+        implements Closeable
 {
-    private ThreadContextClassLoader() {}
```

Comment:
Please see https://github.com/trinodb/trino/pull/8411/files, we don't want one PR to remove something and the restore it. These two operations should be squashed. Simply don't remove it in a first place.

Reply:
Please apply this comment to other changes like that.

Reply:
done, squashed everything into a single commit